ga4gh/datamodel/variants.py

```diff
@@ -619,11 +460,80 @@ def _addDataFile(self, filename):
             chrom, _, _ = self.sanitizeVariantFileFetch(chrom)
             if not isEmptyIter(varFile.fetch(chrom)):
                 if chrom in self._chromFileMap:
-                    raise exceptions.OverlappingVcfException(filename, chrom)
-                self._updateMetadata(varFile)
-                self._updateCallSetIds(varFile)
-                self._chromFileMap[chrom] = filename
-        varFile.close()
+                    raise exceptions.OverlappingVcfException(dataUrl, chrom)
+            self._chromFileMap[chrom] = dataUrl, indexFile
+        self._updateMetadata(varFile)
+        self._updateCallSetIds(varFile)
+        self._updateVariantAnnotationSets(varFile)
+
+    def _updateVariantAnnotationSets(self, variantFile):
+        """
+        Updates the variant annotation set associated with this variant using
+        information in the specified pysam variantFile.
+        """
+        # TODO check the consistency of this between VCF files.
+        if self._variantAnnotationSet is None:
+            annotationType = None
+            infoKeys = variantFile.header.info.keys()
+            if 'CSQ' in infoKeys:
```

Comment:
This isn't correct, there are three cases at the moment. 1kg uses CSQ, and SNPeff and VEP both use ANN, just formatted differently. It's OK to merge with this, just leaving a note.